ossdbtoolsservice/query_execution/query_execution_service.py

```diff
@@ -315,12 +315,7 @@ def _handle_cancel_query_request(self, request_context: RequestContext, params:
                 request_context.send_response(QueryCancelResult(NO_QUERY_MESSAGE))  # TODO: Localize
                 return
 
-            # Only cancel the query if we're in a cancellable state
-            if query.execution_state is ExecutionState.EXECUTED:
-                request_context.send_response(QueryCancelResult('Query already executed'))  # TODO: Localize
-                return
-
-            query.is_canceled = True
+            query.cancel(request_context)
```

Comment:
we already have a function that handles query_cancellation. Line no : 323.

it confusing this way. Keep all the cancellation logic in self.cancel_query itself.

Reply:
The cancel_query function in query_execution_service is needed as it cancels the execution from the server side. i.e runs the kill query on the current connection while cancel() in query cancels the operation inside our service